runtime/core.go

```diff
@@ -893,7 +893,62 @@ func (op compareOp) slot(t *Type) *binaryOpSlot {
 }
 
 func compareRich(f *Frame, op compareOp, v, w *Object) (*Object, *BaseException) {
-	// TODO: Support __cmp__.
+	r, raised := tryRichCompare(f, op, v, w)
+	if raised != nil {
+		return nil, raised
+	}
+	if r != NotImplemented {
+		return r, nil
+	}
+	return try3wayToRichCompare(f, op, v, w)
+}
+
+// convert3wayToObject converts the integer results from a 3-way
+// comparison to a suitable boolean value for the given rich
+// comparison op.
+func convert3wayToObject(op compareOp, c int) *Object {
+	b := false
+	switch op {
+	case compareOpLT:
+		b = c < 0
+	case compareOpLE:
+		b = c <= 0
+	case compareOpEq:
+		b = c == 0
+	case compareOpNE:
+		b = c != 0
+	case compareOpGE:
+		b = c >= 0
+	case compareOpGT:
+		b = c > 0
+	}
+	if b {
```

Comment:
make cover cross references with the lines changed according to git diff. That workflow doesn't work very well since there are typically multiple commits per PR. You can run `make DIFF_COMMIT=xyz cover` to have it cross reference with a different commit but it's a bit awkward. I need to find a nicer way to get coverage into a normal workflow.